uaclient/api/u/pro/auto_attach/with_short_retry/v1.py

```diff
@@ -0,0 +1,243 @@
+import logging
+import sys
+import time
+from typing import List, Optional
+
+from uaclient import (
+    actions,
+    cli,
+    contract,
+    event_logger,
+    exceptions,
+    messages,
+    util,
+)
+from uaclient.api.api import APIEndpoint
+from uaclient.api.data_types import AdditionalInfo
+from uaclient.api.errors import APIError
+from uaclient.cli import _detach
+from uaclient.clouds import AutoAttachCloudInstance  # noqa: F401
+from uaclient.clouds import identity
+from uaclient.config import UAConfig
+from uaclient.data_types import (
+    BoolDataValue,
+    DataObject,
+    Field,
+    StringDataValue,
+    data_list,
+)
+from uaclient.entitlements.entitlement_status import CanEnableFailure
+from uaclient.files import UAFile
+
+event = event_logger.get_event_logger()
+
+
+class CloudError(APIError):
+    pass
+
+
+class AutoAttachDisabledError(APIError):
+    pass
+
+
+class DetachError(APIError):
+    pass
+
+
+class AutoAttachError(APIError):
+    pass
+
+
+class ProAlreadyAttachedError(APIError):
+    pass
+
+
+class AutoAttachWithShortRetryOptions(DataObject):
+    fields = [
+        Field("enable", data_list(StringDataValue), False),
+        Field("enable_beta", data_list(StringDataValue), False),
+    ]
+
+    def __init__(self, enable: List[str], enable_beta: List[str]):
+        self.enable = enable
+        self.enable_beta = enable_beta
+
+
+class AutoAttachWithShortRetryResult(DataObject, AdditionalInfo):
+    fields = [
+        Field("successful", BoolDataValue),
+    ]
+
+    def __init__(self, *, successful: bool):
+        self.successful = successful
+
+
+def get_cloud_instance(cfg: UAConfig) -> Optional[AutoAttachCloudInstance]:
```

Comment:
This seems like the same code we use on `action_auto_attach` on the `cli` module. I think it is better to capture that behavior on the `actions` module instead. In that way, both `cli` and the `api` module can reuse the same code.

Additionally, @orndorffgrant  do you think on this refactoring we should already consider making the `action` function return a `Result` object ?

Reply:
Sorry I missed this. I think the actions module may go away eventually. Its purpose was to be the one true implementation of a given feature, but that role is now filled by the api functions.
So once all of our cli is moved to using the api, then the `actions` module won't need to exist anymore.

The `Result` object is an api specific thing, so I think it might be a bit odd for the `actions` function to return it, but I don't feel strongly about it. I do think it is important to keep our PRs smaller in general, so if we want to refactor `actions` to return API `Results` then I think that should be a separate PR.

Reply:
Agreed, let's not touch on the `actions` module for now